inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -654,12 +653,26 @@ void RemovePermutationsNHWCToNCHWPass::run() {
         auto next = getInputTo(l->outData.front()).begin()->second;
         auto prev = CNNNetPrevLayer(l);
 
-        if (!LayerInfo(next).isPermute() || !LayerInfo(prev).isPermute()) {
+        // The next layer must be NCHW to NHWC permute
+        if (!LayerInfo(next).isPermute() || next->GetParamAsInts("order") != GetPermuteOrder(Layout::NCHW, Layout::NHWC)) {
             continue;
         }
 
-        if (getPassManager()->getPolicy().NHWCToNCHWPolicy == Policy::NHWCToNCHW::REMOVE_ALL) {
-            permutationsToRemove.push_back(prev);
+        // The previous layer must be NHWC to NCHW permute or have 1D data
+        if (LayerInfo(prev).isPermute()) {
+            if (prev->GetParamAsInts("order") != GetPermuteOrder(Layout::NHWC, Layout::NCHW)) continue;
```

Comment:
i would also check for prev layer's layout, in that case you only need to guess order based on output layout
GetPermuteOrder(prev, NCHW) - this more like creating permute layer from code and then checking it's order - this will allow you to create meaningfull errors as well.